include/nccl_ofi_rdma.h

```diff
@@ -602,6 +603,9 @@ struct nccl_net_ofi_rdma_ep {
 	nccl_ofi_freelist_t *bounce_buff_reqs_fl;
 	/* Size of bounce buffers */
 	size_t bounce_buff_size;
+
+	/* True if this ep is stored in the thread-local store */
+	bool thread_local_ep;
```

Comment:
i'm not sure I understand both why this is a property of the ep, nor why this is the right thing to test for.

Reply:
This is a bit thorny. When calling `release_ep`, we can only call `free` on the endpoint reference if it is not stored in the device's thread-local ep store ([explanation](https://github.com/aws/aws-ofi-nccl/blob/aa4d36e549f12a9ca32f4ffb53cbe1e0e74b4e3f/src/nccl_ofi_rdma.c#L5367-L5382)). So, during `release_ep`, we need a way to know whether this is a thread-local endpoint, hence this property of the ep.

I couldn't immediately think of a better way.

Reply:
Is there a different API that's wrong such that we end up in this corner of hell?

Reply:
I suppose the ugliness is because `get_ep` and `release_ep` have the unexpected side effect of tinkering with a thread-local endpoint reference. We still need that behavior for the send communicators. Maybe we can revisit how we deal with the thread-local ep reference after this change.

Reply:
You're suggesting that we add more bad code rather than fix previously created abstraction issues?

Reply:
On abstraction issues, I think we have a couple of problems:
1. Currently (in master today), the API (nccl_ofi_api.c) "owns" the endpoints, and is responsible for creating and destroying the endpoints that are used by the communicators.
2. This PR is attempting to make recv_comm own its endpoint that it uses for communication (RDMA protocol only), but trying to keep compatibility with the API's semantics on endpoint creation and deletion, which needs this special case.

I think fixing this properly would take some refactoring effort to move endpoint ownership into the communicators instead of the API.

One simple thing we can do is set the `release_ep` member of endpoint appropriately, based on whether or not it is a thread-local ep. I.e., we'd have two release_ep methods: `release_ep` and `release_ep_thread_local`, and the `base_ep->release_ep` member function would be set to one of these.

Reply:
When you're designing in the future, you need to take this kind of issue into account; the right answer is to do the required refactoring.  But for now, we'll deal with this.  I really hate the multiple release_ep ideas.